gotime.js

```diff
@@ -1078,11 +1099,35 @@ var fmtParts = {
     return formatTimestampTokens(timestamp, tokenizeFormat(format));
   }
 
+  // Formats an object containing date and time components using the provided format
+  // @param {Object} dateTime Date and time of the form: { date: {year: 2017, month: 10, day: 27}, time: {hour: 14, minute: 15, second: 30}  }
+  // @param {string} format
+  function formatDateTime(dateTime, format) {
+    if(dateTime.time === undefined){
+      dateTime.time = {};
+    }
+    if(dateTime.time.hour === undefined){
+      dateTime.time.hour = 0;
+    }
+    if(dateTime.time.second === undefined){
+      dateTime.time.second = 0;
+    }
+    if(dateTime.time.minute === undefined){
+      dateTime.time.minute = 0;
+    }
+    if(dateTime.time.second === undefined){
```

Comment:
Repeated?

Reply:
👍

Reply:
Is there a better pattern for this?

Reply:
@theothertomelliott We're saying the this block is duplicated on line #1112 (a few above)

Reply:
Done.